src/javascripts/modules/pills.js

```diff
@@ -1,24 +1,53 @@
 export default class Pills {
   constructor(el) {
     this.el = el
+    this.setVars()
     this.setupListeners()
   }
 
+  setVars() {
+    this.ticking = false
+    this.lastScrollY = 0
+  }
+
   setupListeners() {
     window.addEventListener('load', () => this.handleLoad())
+    window.addEventListener('scroll', () => this.handleScroll())
```

Comment:
With this code, we now have two scroll listeners on the window, one for the pillbar and one for the pills in the header/title area. The same goes for the request animation frame stuff below. This provides a good opportunity to refactor things using the Intersection Observer API, but I'd do that work in your next PR.